### PR TITLE
Saeed/add eviction strategy

### LINKED ISSUE #5: Add strategy to deleting keys
like Redis cache I want to add Evic strategy such as lur random and ....
is it support ?
if not i want to code about that.
CC @Alirexaa

### PR DESCRIPTION
**Pull request by SaeedSafi1999**

### CODE REVIEW

File: .config/nextest.toml [hunk 1]
```diff
@@ -1,12 +1,7 @@
 [profile.ci]
-# Retry tests before failing them.
 retries = { backoff = "exponential", count = 2, delay = "2s" }
-# Do not cancel the test run on the first failure.
 fail-fast = false
-# Print out output for failing tests as soon as they fail, and also at the end
-# of the run (for easy scrollability).
 failure-output = "immediate-final"
 
-# save junit report
```
Comment (Alirexaa) [inline .config/nextest.toml]: revert this.

File: src/public_api/server.rs [hunk 2]
```diff
@@ -117,7 +248,7 @@ pub async fn run_server(
             .app_data(web::Data::new(cache.clone()))
             .route("/api/set", web::post().to(set))
             .route("/api/get/{cluster}/{key}", web::get().to(get))
-            .route("/api/ping", web::get().to(check_connection))
+            .route("/api/get/check", web::get().to(check_connection))
```
Comment (Alirexaa) [inline src/public_api/server.rs]: why?

File: src/public_api/server.rs [hunk 3]
```diff
@@ -85,16 +217,15 @@ pub async fn get_all_clusters(
...
     } else {
         HttpResponse::Ok().json(ApiResponse::fail("No clusters found on this port"))
     }
 }
 
-pub async fn check_connection()
- -> HttpResponse {
-  HttpResponse::Ok().json(ApiResponse::ok("PONG"))
+pub async fn check_connection() -> HttpResponse {
+    HttpResponse::Ok().json(ApiResponse::ok("Tock"))
```
Comment (Alirexaa) [inline src/public_api/server.rs]: why?

File: src/public_api/server.rs [hunk 4]
```diff
@@ -1,8 +1,139 @@
```
Comment (Alirexaa) [inline src/public_api/server.rs]: a lot of code comment in bellow why?

File: .config/nextest.toml [hunk 5]
```diff
@@ -7,6 +7,6 @@ fail-fast = false
 # of the run (for easy scrollability).
 failure-output = "immediate-final"
 
-# save junit report
 [profile.ci.junit]
-path = "junit.xml"
\ No newline at end of file
+path = "junit.xml"
+# save junit report
```
Comment (Alirexaa) [inline .config/nextest.toml]: revert

File: src/cache/cache.rs [hunk 6]
```diff
@@ -94,5 +119,184 @@ impl Cache {
...
+    // Eviction strategies
+    fn evict_entries(&self) {
+        let mut store = self.store.lock().unwrap();
+        let mut memory_handler = self.memory_handler.lock().unwrap();
+        let eviction_strategy = EvictionStrategy::from_i32(self.evict_type);
+        match eviction_strategy.unwrap() {
+            EvictionStrategy::VolatileLru => self.evict_volatile_lru(&mut store, &mut memory_handler),
+            EvictionStrategy::AllKeysLru => self.evict_allkeys_lru(&mut store, &mut memory_handler),
+            EvictionStrategy::AllKeysRandom => self.evict_allkeys_random(&mut store, &mut memory_handler),
+            //EvictionStrategy::VolatileRandom => self.evict_volatile_random(&mut store, &mut memory_handler),
```
Comment (Alirexaa) [inline src/cache/cache.rs:146]: why this is commented?
if its unnecessary remove it.

File: src/cache/cache.rs [hunk 7]
```diff
@@ -94,5 +119,184 @@ impl Cache {
...
+    //     if let Some((cluster_key, key_to_evict)) = keys.choose(&mut rand::thread_rng()) {
+    //         if let Some(cluster_store) = store.get_mut(cluster_key) {
+    //             if let Some((value, _, _)) = cluster_store.remove(key_to_evict) {
+    //                 let memory_usage = std::mem::size_of_val(&value);
+    //                 memory_handler.delete_memory(memory_usage);
+    //                 println!("Evicted [{}] from cluster [{}] using volatile random strategy", key_to_evict, cluster_key);
+    //             }
+    //         }
+    //     }
+    // }
```
Comment (Alirexaa) [inline src/cache/cache.rs]: why commented ?

File: src/cache/cache.rs (hunk 6 above)
Comment (SaeedSafi1999) [reply src/cache/cache.rs:146]: no I want fix that later 
I need this scope of codes

File: src/cache/cache.rs (hunk 7 above)
Comment (SaeedSafi1999) [reply src/cache/cache.rs]: I need this code

File: src/cache/cache.rs (hunk 6 above)
Comment (Alirexaa) [reply src/cache/cache.rs:146]: its better to remove unused code.

**Review by Alirexaa — APPROVED** [review]

**Pull request merged**